packages/sources/finage-test/src/endpoint/http/eod.ts

```diff
@@ -0,0 +1,95 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { customSettings } from '../../config'
+
+export const inputParameters = {
+  base: {
+    aliases: ['from', 'coin'],
+    required: true,
+    type: 'string',
+    description: 'The symbol of the currency to query',
+  },
+} as const
+
+interface RequestParams {
+  base: string
+}
+
+interface ResponseSchema {
+  symbol: string
+  totalResults: number
+  error?: string
+  results: [
+    {
+      o: number
+      h: number
+      l: number
+      c: number
+      v: number
+      t: number
+    },
+  ]
+}
+
+type EndpointTypes = {
+  Request: {
+    Params: RequestParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+
+export const httpTransport = new HttpTransport<EndpointTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      return {
+        params: [param],
+        request: {
+          baseURL: config.API_ENDPOINT,
+          url: `/agg/stock/prev-close/${param.base}`,
```

Comment:
I think we can batch on base for this endpoint. Looks like we just send a comma-separated list of bases in v2.

Reply:
Batching may change the ResponseSchema

Reply:
Actually I don't think that batching for this endpoint is working in v2. I tried to send a request with `base: ["AAPL", "MSFT"]` but got error response. Also checking the api it seems they don't have such option, maybe it was removed. 
However finage has new api endpoint for getting all stock data for a given date. We probably can use this to get stock data for previous day.

Reply:
Good catch! I took a look at the api that returns all stock data for the previous day and I think it's too heavy to use here. Looks like it returns 15k+ results. I think we're good to stick to what you have.